### PR TITLE
#3156 - Fixed autocomplete in group results in new activity > post in: group should be sorted by title alphabetically

### LINKED ISSUE #3156: Autocomplete in group results in new activity > post in: group should be sorted by title alphabetically
**Describe the bug**
When posting new activity and selecting a post in: group, autocomplete suggestions for groups should be sorted out by title alphabetically and not by date created. The reason is that there is an instance where a keyword used to search is also in another group's description. It will make users confused and will think that the group they expect to appear to won't be there. And also, the groups shown are only limited to 2 results which are too few. 

**To Reproduce**
1. Create at least 3 groups.
e.g. 
  1st - titled "Group test A" with no description
  2nd - titled "Group test B" with description "Group test A"
  3rd - titled "Group test C" with description "Group test A"
2. Go to the news feed and create one.
3. Select **post in: group** and try searching for "Group test A"
4. You will see that groups will appear only **Group test B** and **Group test C** because Group test C and B were created later than **Group test A**

**Expected behavior**
Normally, autocomplete suggestions should be sorted out by the title. Mostly, users search based on title and I think per_page should at least be 5.

**Screenshots**
image
https://www.loom.com/share/2ca50b34cfe64796a4cdc26cef0ffb1d

**Environment**
Which site did you replicate this issue on?
Replicated locally 

**Support ticket links**
https://secure.helpscout.net/conversation/1668799869/169944/

### PR DESCRIPTION
**Pull request by kent-carlo**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #3156

### How to test the changes in this Pull Request:
https://www.loom.com/share/2ca50b34cfe64796a4cdc26cef0ffb1d

### Proof Screenshots or Video

image

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Added code to search group autocomplete results so will be sorted alphabetically by default and changed per page to 5

### CODE REVIEW

File: src/bp-templates/bp-nouveau/includes/activity/ajax.php [hunk 1]
```diff
@@ -519,8 +519,10 @@ function bp_nouveau_ajax_get_activity_objects() {
 				'user_id'      => bp_loggedin_user_id(),
 				'search_terms' => $_POST['search'],
 				'show_hidden'  => true,
-				'per_page'     => 2,
+				'per_page'     => 5,
 				'exclude'      => $exclude_groups,
+				'orderby'	   => 'title',
+				'order'	       => 'ASC'
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/ajax.php:525]: Follow proper code alignment and spacing

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need changes

File: src/bp-templates/bp-nouveau/includes/activity/ajax.php [hunk 2]
```diff
@@ -519,8 +519,10 @@ function bp_nouveau_ajax_get_activity_objects() {
 				'user_id'      => bp_loggedin_user_id(),
 				'search_terms' => $_POST['search'],
 				'show_hidden'  => true,
-				'per_page'     => 2,
+				'per_page'     => 5,
 				'exclude'      => $exclude_groups,
+				'orderby'      => 'title',
+				'order'	       => 'ASC'
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/ajax.php:525]: I think you can use the **bp_before_groups_get_groups_parse_args** filter to make the changes and add the **orderby** and **order** parameters into the query. Maybe we can't change globally.

**Review by chetansatasiya — APPROVED** [review]

**Review by chetansatasiya — APPROVED** [review]

**Review by dipskakadiya — APPROVED** [review]

**Comment by gabybeldoro** [comment]
Retested in the QA ongoing branch
Status: Fixed

Proof Screenshots or Video:
https://www.screencast.com/t/B75nUEjNRP

**Comment by joeytoribio** [comment]
Hi @kent-carlo,

This has been tested in the qa-ongoing branch without any issues.

**TC Title:** 
New Activity > Post in; Group - Suggestion result is not filtered 

**TC:** 
To Reproduce
Steps to reproduce the behavior:

1] Create at least 3 groups. e.g.
    1st - titled "Group test A" with no description
    2nd - titled "Group test B" with description "Group test A"
    3rd - titled "Group test C" with description "Group test A"
2] Go to the news feed and create one.
3] Select post in: group and try searching for "Group test A"
4] You will see that groups will appear only Group test B and Group test C because Group test C and B were created later than Group test A

**Video:**
https://www.loom.com/share/b9ad9459c16040eeb4a546cf3fb86c40

**Status: Passed**

**Comment by basil-buddyboss** [comment]
Hello @kent-carlo 

This has been tested in the QA-approved branch and there were no issues encountered.

Result: Pass

**Comment by sourabhmatolia** [comment]
**UPDATE**

Groups per page and order issue will be fixed in Post Form task here:
https://github.com/buddyboss/buddyboss-platform/pull/3190

Closing this PR. Thanks.

**Pull request closed without merging**